src/Voron/Impl/EncryptionBuffersPool.cs

```diff
@@ -106,5 +109,14 @@ public void Dispose()
             {
             }
         }
+
+        public void LowMemory()
+        {
+            ReleaseUnmanagedResources();
```

Comment:
Just checking, but is this thread safe?

Reply:
It should be. The _items array is not written to at any point after construction, and the internal stacks are ConcurrentStack, so they should be able to handle multiple threads going through the stacks and poping elements out.